cmd/earthly/main.go

```diff
@@ -2801,6 +2802,51 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 			return errors.Wrapf(err, "parse target name %s", targetName)
 		}
 	}
+
+	cleanCollection := cleanup.NewCollection()
+	defer cleanCollection.Close()
+
+	cc, err := cloud.NewClient(app.apiServer, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	// Default upload logs, unless explicitly configured
+	if !app.cfg.Global.DisableLogSharing {
+		_, _, _, err := cc.WhoAmI()
+		isLoggedIn := err == nil
+
+		if isLoggedIn {
+			// If you are logged in, then add the bundle builder code, and configure cleanup and post-build messages.
+			app.console = app.console.WithLogBundleWriter(target.String(), cleanCollection)
+
+			defer func() { // Defer this to keep log upload code together
+				logPath, err := app.console.WriteBundleToDisk()
+				if err != nil {
+					app.console.Warnf(err.Error())
+					return
+				}
+
+				id, err := cc.UploadLog(logPath)
+				if err != nil {
+					app.console.Warnf(err.Error())
+					return
+				}
+				app.console.Printf("Share your build log with this link: %s\n", id)
+			}()
+		} else {
+			// If you are not logged in, then advertise the service, since they probably turned it on to try it.
+			defer func() { // Defer this to keep log upload code together
+				switch err {
+				case cloud.ErrUnauthorized:
+					app.console.Printf("Share your logs with an Earthly account! Register for one at https://ci.earthly.dev.")
+				default:
+					app.console.Warnf("Logs for this run were not shared, since earthly couldn't login. Error: %s", err.Error())
```

Comment:
Should we make this more concise? 
```suggestion
					app.console.Warnf("Logs were not shared, due to earthly login error: %s", err.Error())
```

Reply:
I'll leave wording up to those with stronger opinions. If nobody else has suggestions, I will use this.

Reply:
@vladaionescu thoughts on the reword?

Reply:
Looks good to me

Reply:
Wording changed.